data/flow/flowdata.py

```diff
@@ -281,36 +310,47 @@ def _parse_function(example_proto):
 
             return image_a, image_b, flow
 
-        dataset = tf.data.TFRecordDataset(filenames)
+        files = tf.data.Dataset.list_files(filename_pattern, shuffle=True)
+        dataset = tf.data.TFRecordDataset(files, compression_type='GZIP', num_parallel_reads=min(4, self.batch_size))
         if repeat:
-            dataset = dataset.apply(tf.contrib.data.shuffle_and_repeat(buffer_size=self.batch_size * 10))
-        else:
-            dataset = dataset.shuffle(buffer_size=self.batch_size * 10)
+            dataset = dataset.repeat()
         dataset = dataset.map(_parse_function, num_parallel_calls=multiprocessing.cpu_count())
         dataset = dataset.batch(self.batch_size)
         dataset = dataset.prefetch(2)
         return dataset
 
 
-def _write_shard(shard_id, shard_range, image_a_paths, image_b_paths, flow_paths, filename, directory, verbose):
+def _write_shard(shard_id, shard_range, image_a_paths, image_b_paths, flow_paths, filename, directory, verbose,
+                 max_flow):
     """
     :param shard_id: Index of the shard.
     :param shard_range: Iteration range of the shard.
     :param image_paths: Path of all images.
     :param flow_paths: Path of all flows.
     :param filename: Base name of the output shard.
     :param directory: Output directory.
+    :param verbose: Whether to print to console.
+    :param max_flow: Float. Maximum flow magnitude of the flow image. Any examples with flow magnitude greater than this
+        will be ignored.
     :return: Nothing.
     """
     if verbose and len(shard_range) > 0:
         print('Writing to shard', shard_id, 'data points', shard_range[0], 'to', shard_range[-1])
 
-    writer = tf.python_io.TFRecordWriter(os.path.join(directory, str(shard_id) + '_' + filename))
+    record_name = os.path.join(directory, str(shard_id) + '_' + filename)
+    options = tf.python_io.TFRecordOptions(tf.python_io.TFRecordCompressionType.GZIP)
```

Comment:
How are the savings on disk with this compression, roughly speaking?

Reply:
Good savings. 10-20% disk space saved. Given that we are working with shared network drives in the cluster, I believe that this helps.